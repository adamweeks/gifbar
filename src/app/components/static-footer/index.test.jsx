import React from 'react';
import renderer from 'react-test-renderer';

import StaticFooter from '.';

describe(`StaticFooter component`, () => {
  /**
    count: PropTypes.number,
    changeOffset: PropTypes.func,
    currentOffset: PropTypes.number,
    isTrending: PropTypes.bool,
    navPrevEnabled: PropTypes.bool,
    navForwardEnabled: PropTypes.bool,
    showNav: PropTypes.bool,
    totalResults: PropTypes.number,
   */



  it(`renders properly trending`, () => {
    const component = renderer.create(
      <StaticFooter
        isTrending
      />
    );
    expect(component).toMatchSnapshot();
  });

  describe(`show nav`, () => {
    it(`renders properly`, () => {
      const component = renderer.create(
        <StaticFooter
          count={10}
          changeOffset={jest.fn()}
          currentOffset={0}
          navPrevEnabled
          navForwardEnabled
          showNav
          totalResults={100}
        />
      );
      expect(component).toMatchSnapshot();
    });

    it(`renders properly with prev disabled`, () => {
      const component = renderer.create(
        <StaticFooter
          count={10}
          changeOffset={jest.fn()}
          currentOffset={0}
          navForwardEnabled
          showNav
          totalResults={100}
        />
      );
      expect(component).toMatchSnapshot();
    });

    it(`renders properly with forward disabled`, () => {
      const component = renderer.create(
        <StaticFooter
          count={10}
          changeOffset={jest.fn()}
          currentOffset={0}
          navPrevEnabled
          showNav
          totalResults={100}
        />
      );
      expect(component).toMatchSnapshot();
    });
  })

});
