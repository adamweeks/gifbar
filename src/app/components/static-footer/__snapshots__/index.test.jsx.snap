// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticFooter component renders properly trending 1`] = `
<div
  className="footer"
>
  <div
    className="side"
  />
  <div
    className="main"
  >
    <div
      className="title"
    >
      Top Trending GIFs!
    </div>
    <div
      className="attribution"
    >
      <img
        className="img"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="side"
  >
    <div
      className="button"
      onClick={undefined}
    >
      <img
        className="imgRefresh"
        src="test-file-stub"
      />
    </div>
  </div>
</div>
`;

exports[`StaticFooter component show nav renders properly 1`] = `
<div
  className="footer"
>
  <div
    className="side"
  >
    <div
      className="button"
      onClick={[Function]}
    >
      &lt;
    </div>
  </div>
  <div
    className="main"
  >
    <div
      className="title"
    >
      Showing page 1 of 10
    </div>
    <div
      className="attribution"
    >
      <img
        className="img"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="side"
  >
    <div
      className="button"
      onClick={[Function]}
    >
      &gt;
    </div>
  </div>
</div>
`;

exports[`StaticFooter component show nav renders properly with forward disabled 1`] = `
<div
  className="footer"
>
  <div
    className="side"
  >
    <div
      className="button"
      onClick={[Function]}
    >
      &lt;
    </div>
  </div>
  <div
    className="main"
  >
    <div
      className="title"
    >
      Showing page 1 of 10
    </div>
    <div
      className="attribution"
    >
      <img
        className="img"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="side"
  >
    <div
      className="button buttonDisabled"
      onClick={[Function]}
    >
      &gt;
    </div>
  </div>
</div>
`;

exports[`StaticFooter component show nav renders properly with prev disabled 1`] = `
<div
  className="footer"
>
  <div
    className="side"
  >
    <div
      className="button buttonDisabled"
      onClick={[Function]}
    >
      &lt;
    </div>
  </div>
  <div
    className="main"
  >
    <div
      className="title"
    >
      Showing page 1 of 10
    </div>
    <div
      className="attribution"
    >
      <img
        className="img"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="side"
  >
    <div
      className="button"
      onClick={[Function]}
    >
      &gt;
    </div>
  </div>
</div>
`;
